external/js/cothority/lib/skipchains/skipchains.js

```diff
@@ -0,0 +1,120 @@
+"use strict";
+
+const net = require("../net");
+const protobuf = require("../protobuf");
+const misc = require("../misc");
+
+const skipchainID = "Skipchain";
+
+class SkipchainClient {
+
+    /**
+     * Returns a new skipchain client from a roster
+     *
+     * @param {cothority.Roster} roster the roster over which the client can talk to
+     * @param {string} last know skipblock ID in hexadecimal format
+     * @returns {SkipchainClient} A client that can talks to the skipchain services
+     */
+    constructor(group, roster,lastID) {
+        this.lastRoster = roster;
+        this.lastID = misc.hexToUint8Array(lastID);
+        this.socket = new net.RosterSocket(this.lastRoster);
+        this.group = group;
+    }
+
+    /**
+     * Returns the latest known skipblockID
+     *
+     * @returns {string} hexadecimal encoded skipblockID
+     */
+    get latestID() {
+        return misc.uint8ArrayToHex(this.lastID);
+    }
+
+    /**
+     * updateChain asks for the latest block of the skipchain with all intermediate blocks.
+     * It automatically verifies the transition from the last known skipblock ID to the
+     * latest one returned. It also automatically save the latest good known
+     * roster from the latest block.
+     * @return {Promise} A promise which resolves with the latest skipblock if
+     * all checks pass.
+     */
+    latestBlock() {
+        const requestName = "GetUpdateChain";
```

Comment:
Sorry, yesterday I changed the behavior of GetUpdateChain in #1034. For now, this is good enough, but to handle evolving rosters, the client needs to check that latest is in fact a terminal, and if not, it needs to keep calling GetUpdateChain on the latest.Roster, to keep following the chain as it moves to other servers.